wikimedia/wikipedia-ios
# Linter: Add SwiftLint (1 of 3)

**Phabricator:** https://phabricator.wikimedia.org/T308566

### Notes
This PR adds SwiftLint for linting and code formatting our Swift code. It includes:

- Updating existing Homebrew setup script to install `swiftlint`
- A build script phase that lints and formats on every build for our app targets
- Separate autocorrected by code formatter and linter warnings/errors YAML configuration files. While the contents are currently the same, over time these can diverge based on our coding style desires.

The disabled rules in the YAML files are required to build the current project with no warnings or errors. The subsequent PRs slowly chip away at these disabled rules, both the ones that require manual intervention and ones that can be automatically corrected by SwiftLint. As discussed previously, we can remove more over time, but for now I feel over the course of these 3 PRs I've put together a sensible starting point.

### Test Steps
Confirm the app targets build before and after installing SwiftLint (manually or via the script).
